api/services/MachineService.js

```diff
@@ -475,17 +492,26 @@ function _updateMachinesPool() {
  * @return {null}
  */
 function _shouldTerminateMachine(machine) {
-  machine.isSessionActive()
-    .then((isActive) => {
+  Promise.props({
+    isActive: machine.isSessionActive(),
+    config: ConfigService.get('neverTerminateMachine')
+  })
+    .then((props) => {
```

Comment:
You should use object destructuring in this case: https://ponyfoo.com/articles/es6-destructuring-in-depth